instrumentation/apache-pulsar/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/telemetry/PulsarTelemetry.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.telemetry;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessageOperation;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessagingAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessagingAttributesGetter;
+import io.opentelemetry.instrumentation.api.internal.InstrumenterUtil;
+import io.opentelemetry.javaagent.instrumentation.pulsar.VirtualFieldStore;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.time.Instant;
+import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+
+public final class PulsarTelemetry {
+  private static final String INSTRUMENTATION = "io.opentelemetry:pulsar-client";
+  private static final OpenTelemetry TELEMETRY = GlobalOpenTelemetry.get();
+  private static final TextMapPropagator PROPAGATOR =
+      TELEMETRY.getPropagators().getTextMapPropagator();
+
+  private static final SpanNameExtractor<Message<?>> CONSUMER_RECEIVE =
+      new InternalSpanNameExtractor<>(SpanKind.CONSUMER, MessageOperation.RECEIVE);
```

Comment:
how about `MessagingSpanNameExtractor` to follow messaging semantic conventions?